ceno_host/src/lib.rs

```diff
@@ -127,8 +127,10 @@ impl CenoStdin {
     pub fn write(
         &mut self,
         item: &impl for<'a> Serialize<HighSerializer<AlignedVec, ArenaHandle<'a>, Error>>,
-    ) -> Result<(), Error> {
-        to_bytes::<Error>(item).map(|bytes| self.write_slice(bytes))
+    ) -> Result<&mut Self, Error> {
+        to_bytes::<Error>(item)
+            .map(|bytes| self.write_slice(bytes))
+            .map(|_| self)
```

Comment:
```suggestion
        let bytes = to_bytes::<Error>(item)?;
        self.write_slice(bytes);
        Ok(self)
```

Let's be more explicit.